rpc/src/v1/impls/private.rs

```diff
@@ -90,7 +90,12 @@ impl Private for PrivateClient {
 			data: Some(transaction.data.into()),
 			condition: None,
 		};
+		// TODO [ToDr] This is invalid?
+		// private-tx/src/lib.rs#L532 uses engine to figure out the scheme.
 		let (contract_address, _) = contract_address(CreateContractAddress::FromSenderAndNonce, &signed_transaction.sender().clone(), &transaction.nonce.into(), &[]);
+		// TODO [ToDr] Figure out this:
+		// Why this type contains the entire transaction (request)?
+		// Why Request and not entire SignedTransaction?
```

Comment:
I assume, that the user will sign the transaction by himself

Reply:
But that's the output type, right? We already had signed transaction after `client.public_creation_transaction` and the we disregard the signature and return just the request? So why sign it in the first place?

Reply:
So may be I don't understand your question. Let me show this process, basing on our test (see more comments and explanations inline as additional comments):

trace!("Creating private contract");
// Create an input for deployment transaction. The input basically contains private contract in data
let private_contract_test = "606060...79d2080029".from_hex().unwrap();
let mut private_create_tx = Transaction::default();
private_create_tx.action = Action::Create;
private_create_tx.data = private_contract_test;
let private_create_tx_signed = private_create_tx.sign(&key1.secret(), None);
// The input is created. Pass it and additional parameters to API in order to create a deployment transaction for the contract from the input
let validators = vec![key3.address(), key4.address()];
let public_tx = pm.public_creation_transaction(BlockId::Latest, &private_create_tx_signed, &validators, 0.into()).unwrap();
// Deployment transaction returned, in order to deploy it in the chain we need to sign it as well
let public_tx = public_tx.sign(&key1.secret(), chain_id);
trace!("Transaction created. Pushing block");
push_block_with_transactions(&client, &[public_tx]);